signal_scanner_bot/twitter.py

```diff
@@ -35,16 +36,73 @@ def get_api():
 def send_tweet(tweet: str, timestamp: datetime, api: tweepy.API) -> None:
     hashtags = " ".join(SEND_HASHTAGS)
-    if len(tweet + hashtags) >= 260:
-        # TODO: better
-        log.warning(f"Cannot tweet message, exceeds length: {tweet}")
-        return
-
-    formatted = dedent(
-        f"""
-    {tweet} @ {timestamp.strftime('%l:%M:%S%p').strip()}
-
-    {hashtags}
-    """
-    )
-    api.update_status(formatted)
+
+    # Check if tweet is longer than 280 minus a 20 char padding, add to list if
+    # not, continue with dividing up text if so
+    if len(tweet + hashtags) >= TWEET_MAX_SIZE - TWEET_PADDING:
+
+        # Split tweet into word list, create empty list to store serialized tweets, index to track last position
+        tweet_word_list = tweet.split(" ")
+        tweets_list = []
+        j = 0
+
+        # For loop through word list and enumerate the index because we'll need
+        # it but don't actually care about the list value so send to null
+        for index, _ in enumerate(tweet_word_list):
+
+            # Check if it is the first tweet, which will contain hashtags and
+            # timestamp. If not first it will only contain the text plus ellipses
+            # and for the final one drop the ellipses
+            if len(tweets_list) == 0:
+                sub_tweet = " ".join(tweet_word_list[j:index]) + " ..." + hashtags
+            elif index < len(tweet_word_list):
+                sub_tweet = " ".join(tweet_word_list[j:index]) + " ..."
+            elif index == len(tweet_word_list):
+                sub_tweet = " ".join(tweet_word_list[j:index])
+
+            # When length of tweet reaches >260 chars save to list and set
+            # base index for next tweet
+            if len(sub_tweet) > TWEET_MAX_SIZE - TWEET_PADDING:
+                tweets_list.append(" ".join(tweet_word_list[j:index - 1]) + " ...")
+                j = index - 1
+
+        # Save the last tweet
+        tweets_list.append(" ".join(tweet_word_list[j:]))
+
+        # Append the tweet number / tweet thread length to end of tweet. Again
+        # don't actually care about the list value so sending to null.
+        for index, _ in enumerate(tweets_list):
+            tweets_list[index] += f" {index + 1}/{len(tweets_list)}"
+
+    else:
+        tweets_list = [tweet]
+
+    # tweet_id used to track previous tweet in thread, set to None for first
+    # This iterates through all (even if just one) tweets in list to send
+    tweet_id = None
+    for index, sub_tweet in enumerate(tweets_list):
+
+        # If this is the first tweet add timestamp and hashtags
+        if index == 0:
+            formatted = dedent(
+                f"""
+            {timestamp.strftime('%H:%M:%S%p').strip()}
+
+            {sub_tweet}
+
+            {hashtags}
+            """
+            )
+        else:
+            formatted = dedent(
+                f"{sub_tweet}"
+            )
```

Comment:
I think this can just be the following:
```suggestion
            formatted = sub_tweet
```